aether-kernel/aether/kernel/api/validators.py

```diff
@@ -83,37 +83,9 @@ def wrapper_validate_schema_input_definition(data):
 
 
 def validate_entity_project(validated_data):
-    from .models import Project
-
     _schema_decorator = validated_data.get('schemadecorator')
-    _submission = validated_data.get('submission')
-    _mapping = validated_data.get('mapping')
-
-    _possible_project = validated_data.get('project')
-    if not _possible_project:
-        if _schema_decorator:
-            _possible_project = Project.objects.filter(schemadecorators__pk=_schema_decorator.pk).first()
-        elif _submission:
-            _possible_project = Project.objects.filter(submissions__pk=_submission.pk).first()
-        elif _mapping:
-            _possible_project = Project.objects.filter(mappings__pk=_mapping.pk).first()
-
-    _artefacts_in_same_project = True
-    if _possible_project:
-        _artefact_dict = {
-            'schema_decorators': str(_schema_decorator.pk) if _schema_decorator else None,
-            'submissions': str(_submission.pk) if _submission else None,
-            'mappings': str(_mapping.pk) if _mapping else None,
-        }
-        _artefacts_in_same_project = in_same_project_and_cache(_artefact_dict, _possible_project)
-    else:
+    if not _schema_decorator:
         raise ValidationError(
-            _('No associated project. Check you provided the correct Submission, Mapping and Schema Decorator')
+            _('Schema Decorator MUST be provided with entities')
         )
-
-    if not _artefacts_in_same_project:
-        raise ValidationError(
-            _('Submission, Mapping and Schema Decorator MUST belong to the same Project')
-        )
-
-    return _possible_project
+    return _schema_decorator.project
```

Comment:
you still need to check that the rest of provided artefacts belong to the same project...

Reply:
I mean, the schema decorator is king but if the submission or mapping are present you also need to check that they belong to the same project as the schema decorator.

```python
        _artefact_dict = {	
            'schema_decorators': str(_schema_decorator.pk),	
            'submissions': str(_submission.pk) if _submission else None,	
            'mappings': str(_mapping.pk) if _mapping else None,	
        }	
        if not in_same_project_and_cache(_artefact_dict, _schema_decorator.project):
            raise ValidationError(	
                _('Submission, Mapping and Schema Decorator MUST belong to the same Project')	
            )
```

Reply:
or simpler
```python
if _submission and _submission.project != _schema_decorator.project:
    raise ValidationError(_('Submission MUST belong to the same Project as Schema Decorator'))

if _mapping and _mapping.project != _schema_decorator.project:
    raise ValidationError(_('Mapping MUST belong to the same Project as Schema Decorator'))
```